### ISSUE TITLE
Read JSON and materialize to plain Julia type

### ISSUE THREAD

**Issue by aplavin** [quinnj/JSON3.jl#215]
It would be useful to provide a generic function, let's call it `materialize` for now, that always makes a plain Julia type from the result of `JSON3.read`. Currently, `copy` is close to being that function, but not quite:
```
julia> read_plain(str) = JSON3.read(str) |> copy

# ok:
julia> read_plain("[1, 2]")
2-element Vector{Int64}:
 1
 2

# not ok:
julia> read_plain("\"1\"")
ERROR: MethodError: no method matching copy(::String)
```

**Comment by quinnj** [comment]
Yeah, good catch. Part of the issue is that `copy` isn't defined by default for immutables in Base. I think there's been some discussion on defining that though recently. In the mean time, I'd be fine defining a `JSON3.materialize` function that does what you're after. It'd be a pretty simple definition, if you'd like to put up a PR!
